### PR TITLE
hotfix: Handle large option sets

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImplKt.kt [hunk 1]
```diff
@@ -311,10 +313,30 @@ class SearchRepositoryImplKt(
...
+                                            metadataIconProvider(
+                                                option.style(),
+                                                program?.style()?.color()?.toColor()
+                                                    ?: SurfaceColor.Primary,
+                                            ),
+                                        )
+                                    }
+                                },
+                        )
+                        /*OptionSetConfiguration.config(
```
Comment (mmmateos) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImplKt.kt:333]: commented code

File: form/src/main/java/org/dhis2/form/model/FieldUiModelImpl.kt [hunk 2]
```diff
@@ -61,13 +61,13 @@ data class FieldUiModelImpl(
 
     override fun invokeUiEvent(uiEventType: UiEventType) {
         callback?.intent(FormIntent.OnRequestCoordinates(uid))
-        if (uiEventType != UiEventType.QR_CODE &&
+        /*if (uiEventType != UiEventType.QR_CODE &&
```
Comment (mmmateos) [inline form/src/main/java/org/dhis2/form/model/FieldUiModelImpl.kt:64]: commented code

File: form/src/main/java/org/dhis2/form/model/OptionSetDialogViewModel.kt [hunk 3]
```diff
@@ -37,8 +37,10 @@ class OptionSetDialogViewModel(
             searchOptionSetOption(
                 field.optionSet,
                 textToSearch,
-                field.optionSetConfiguration?.optionsToShow ?: emptyList(),
-                field.optionSetConfiguration?.optionsToHide ?: emptyList(),
+                /*field.optionSetConfiguration?.optionsToShow ?:*/
```
Comment (mmmateos) [inline form/src/main/java/org/dhis2/form/model/OptionSetDialogViewModel.kt:40]: commented code

File: form/src/main/java/org/dhis2/form/ui/event/UiEventFactoryImpl.kt [hunk 4]
```diff
@@ -44,7 +30,7 @@ class UiEventFactoryImpl(
         var uiEvent: RecyclerViewUiEvents? = null
         try {
             uiEvent = when (uiEventType) {
-                DATE_TIME -> {
+                /*DATE_TIME -> {
```
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/ui/event/UiEventFactoryImpl.kt:33]: This commented code is no longer needed? or was just for development purposes?

Comment (Balcan) [reply form/src/main/java/org/dhis2/form/ui/event/UiEventFactoryImpl.kt:33]: Not used at all :)

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**